### ISSUE TITLE
Model deep-fetch is very slow

### ISSUE THREAD

**Issue by shalstvedt** [AskNature/AN#59]

**Comment by ethansmith** [comment]
out of curiosity, what's an example of a query that takes a long time?

**Comment by shalstvedt** [comment]
Everything tied to a status, for example, takes much, much longer than it should. This will be fixed with the new fetch mechanism.

**Comment by shalstvedt** [comment]
All of these fetches should be on the order of milliseconds, but I've been seeing strategy queries take ~1.5s

**Comment by ethansmith** [comment]
i noticed the same thing with status .. and especially userstatus. i came to a realization last night, but can't remember what it was ;) I'll post here if I manage to recall.

**Comment by shalstvedt** [comment]
hah, ok. a realization about the cause?

**Comment by ethansmith** [comment]
Yes .. and I *think* I may have nailed it down in the branch I'm working on: in the listoptions.js server file I identified the object keys for ContentStatus and UserStatus as strings.

**Comment by ethansmith** [comment]
Actually, I don't think that was the issue. However, I just built hash indexes for UserStatus and ContentStatus and that seems to have eliminated the lag.
